src/components/NicknameForm.tsx

```diff
@@ -0,0 +1,99 @@
+import { InputAdornment, TextField, Typography, Button } from '@mui/material'
+import 'react-toastify/dist/ReactToastify.css'
+import { useForm, SubmitHandler } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { FaAngleRight } from 'react-icons/fa6'
+import { useState, useEffect } from 'react'
+import { toast } from 'react-toastify'
+import { SettingFormsProps } from '~/lib/types'
+
+const nicknameSchema = z.object({
+	nickname: z
+		.string()
+		.min(1, { message: 'Your nickname cannot be empty.' })
+		.regex(/^[a-zA-Z]/, { message: 'Start your name with a letter.' })
+		.min(5, { message: 'Make sure your name is at least 5 characters long.' })
+		.max(15, { message: 'Keep your name within 15 characters.' }),
+})
+
+interface NicknameFormInput {
+	nickname: string
+}
+
+const NicknameForm = ({ setIsDetailsVisible }: SettingFormsProps) => {
+	const [isFocusedField, setIsfocusedField] = useState(false)
+	const {
+		register,
+		handleSubmit,
+		reset,
+		formState,
+		formState: { errors },
+		clearErrors,
+	} = useForm<NicknameFormInput>({
+		resolver: zodResolver(nicknameSchema),
+	})
+
+	const onSubmit: SubmitHandler<NicknameFormInput> = data => {
+		console.log(data)
+	}
+
+	useEffect(() => {
+		if (formState.isSubmitSuccessful) {
+			setIsfocusedField(false)
+			toast.success('Nickname updated successfully')
+			reset()
+		}
+	}, [formState, reset])
```

Comment:
Instead of calling the success toast inside the `useEffect` hook, I'd recommend moving it to the `onSuccess` callback of mutation hook (if you're gonna use react-query)

Reply:
`useEffect `is used temporarily, when I will start creating database integration I intend to change it of course, probably with` tRPC`